lib/src/main/java/growthbook/sdk/java/FeatureEvaluator.java

```diff
@@ -169,48 +303,64 @@ public <ValueType> FeatureResult<ValueType> evaluateFeature(String key, GBContex
-                        .phase(rule.getPhase())
-                        .seed(rule.getSeed())
-                        .hashVersion(rule.getHashVersion())
-                        .filters(rule.getFilters())
-                        .build();
 
-                ExperimentResult<ValueType> result = experimentEvaluator.evaluateExperiment(experiment, context, key);
-                if (result.getInExperiment() && (result.getPassThrough() == null || !result.getPassThrough())) {
-                    ValueType value = (ValueType) GrowthBookJsonUtils.unwrap(result.getValue());
-
-                    FeatureResult<ValueType> experimentFeatureResult = FeatureResult
-                            .<ValueType>builder()
-                            .value(value)
-                            .source(FeatureResultSource.EXPERIMENT)
-                            .experiment(experiment)
-                            .experimentResult(result)
-                            .build();
-
-                    if (featureUsageCallback != null) {
-                        featureUsageCallback.onFeatureUsage(key, experimentFeatureResult);
+                    goOutFromCircularLoop();
+                    return forcedRuleFeatureValue;
+                } else {
+
+                    ArrayList<ValueType> variations = rule.getVariations();
+                    if (variations != null) {
+
+                        // Experiment rule
+                        String experimentKey = rule.getKey();
+                        if (experimentKey == null) {
+                            experimentKey = key;
+                        }
+
+                        // For experiment rules, run an experiment
+                        Experiment<ValueType> experiment = Experiment
+                                .<ValueType>builder()
+                                .key(experimentKey)
+                                .coverage(rule.getCoverage())
+                                .weights(rule.getWeights())
+                                .hashAttribute(rule.getHashAttribute())
+                                .namespace(rule.getNamespace())
+                                .variations(rule.getVariations())
+                                .meta(rule.getMeta())
+                                .ranges(rule.getRanges())
+                                .name(rule.getName())
+                                .phase(rule.getPhase())
+                                .seed(rule.getSeed())
+                                .hashVersion(rule.getHashVersion())
+                                .filters(rule.getFilters())
+                                .variations(variations)
+                                .minBucketVersion(rule.getMinBucketVersion())
+                                .bucketVersion(rule.getBucketVersion())
+                                .disableStickyBucketing(rule.getDisableStickyBucketing())
+                                .fallbackAttribute(rule.getFallbackAttribute())
```

Comment:
Should'nt we also set the ConditionalJson here?

Reply:
Thank you for mentioning this issue. We have fixed it and are now creating a new pull request with these changes.